invenio_users_resources/records/mappings/os-v1/users/user-v2.0.0.json

```diff
@@ -75,10 +96,22 @@
         "type": "boolean"
       },
       "username": {
-        "type": "keyword"
+        "type": "text",
+        "analyzer": "ngram_3_to_5_analyzer",
+        "fields": {
+          "keyword": {
+            "type": "keyword"
+          }
+        }
       },
       "email": {
-        "type": "keyword"
+        "type": "text",
+        "analyzer": "ngram_3_to_5_analyzer",
+        "fields": {
+          "keyword": {
+            "type": "keyword"
+          }
+        }
```

Comment:
Note: this has to be tested and checked through the code as now to exactly match the email it will be email.keyword instead of email

Reply:
in principle when we do that e.g custom fields, we use the `field.keyword` only for faceting.

Reply:
in general, shouldn't we use email tokenizer? I think it is meant for this purpose https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-uaxurlemail-tokenizer.html

Reply:
Now using the email tokenizer as the [default search analyzer](https://opensearch.org/docs/latest/analyzers/search-analyzers/#specifying-the-default-search-analyzer-for-an-index)